spec/features/participant-declarations/submit_participant_declarations_spec.rb

```diff
@@ -105,9 +105,12 @@ def given_an_npq_participant_has_been_entered_onto_the_dfe_service
     create(:schedule, name: "ECF September standard 2021")
     npq_lead_provider = create(:npq_lead_provider, cpd_lead_provider: @cpd_lead_provider)
     npq_course = create(:npq_course, identifier: "npq-leading-teaching")
-    npq_validation_date = create(:npq_validation_data, npq_lead_provider: npq_lead_provider, npq_course: npq_course)
-    @npq_id = npq_validation_date.user.id
-    travel_to npq_validation_date.profile.schedule.milestones.first.start_date + 1.day
+    npq_validation_data = create(:npq_validation_data, npq_lead_provider: npq_lead_provider, npq_course: npq_course)
+    @npq_id = npq_validation_data.user.id
+
+    NPQ::CreateOrUpdateProfile.new(npq_validation_data: npq_validation_data).call
```

Comment:
I wonder if that will result in random percy failures - it seems to be adding faker-looking data to participant details that we show.

Reply:
There are failures. I don't actually understand Percy. mainline has data that should be present missing

Reply:
Huh, it seems the percy snapshots which changed are from percy, `spec/cypress/integration/admin/ParticipantManagment.feature`. Which runs a scenario you didn't touch at all in this PR.

I think removing `after_save :update_participant_profile` hook from npq_validation_data means that the syncing logic is not run in the scenario, and that surfaces Faker randomness. Uhh. 

`spec/cypress/app_commands/scenarios/admin/school_participants.rb` is the scenario in question. I think you want to change 

```
npq_profile = FactoryBot.create :participant_profile,
                                :npq,
                                validation_data: nil,
                                user: npq_user,
                                created_at: Date.parse("19/09/2019")

FactoryBot.create :npq_validation_data,
                  id: npq_profile.id,
                  user: npq_user,
                  date_of_birth: Date.parse("10/12/1982"),
                  nino: "NI123456",
                  created_at: Date.parse("19/09/2019"),
                  teacher_reference_number: "9780824"
```
to be something like 
```
validation_data = FactoryBot.create :npq_validation_data,
                  id: npq_profile.id,
                  user: npq_user,
                  date_of_birth: Date.parse("10/12/1982"),
                  nino: "NI123456",
                  created_at: Date.parse("19/09/2019"),
                  teacher_reference_number: "9780824",
                  urn: "123456"
NPQ::CreateOrUpdateProfile.new(npq_validation_data: validation_data).call
```

Reply:
Ok, I've stuck another commit on the end roughly based on your suggestion